### PR TITLE
Update ci dox

### LINKED ISSUE #2421: Update documentation on CI
The documentation on CI is still about Travis. It should be updated to GitHub Actions. I'm happy to do that when I'm back from vacation.

### LINKED ISSUE #2678: Remove or update charm 6.8 containers
* containers/Dockerfile.buildenv1804 installs Charm 6.8
* .travis/MacOsBuildDependencies.sh takes the version as an argument, but applies a patch from support/Charm which only has 6.7 and 6.8.  (These patches should also go.)

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Update CI dox to GitHub actions
- Remove Travis CI files. We've used GitHub Actions very successfully for a long time now.

Fixes #2678 together with #4019 Each fixes half the issue

Fixes #2421

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: docs/DevGuide/GitHubActions.md [hunk 1]
```diff
@@ -136,27 +135,36 @@ useful to perform at least the following tests locally:
...
-for your pull request.
+- The `gcc Debug` build runs code coverage for each GitHub Actions build.
+
+## Troubleshooting {#github-actions-troubleshooting}
+
+* Occasionally, a build job will fail because of a problem with GitHub Actions
+  (e.g. it times out).  On the `Checks` tab you can restart all or only the
+  failed jobs. In the top right corner there's a `Re-run jobs` menu, which also
+  has `Re-run failed jobs`. This button is `Cancel workflow` during the build
+  process.
```
Comment (knelli2) [inline docs/DevGuide/GitHubActions.md]: I think this is only for core-developers. I don't see the `Re-run jobs` menu or `Cancel workflow` for my PRs. Same for the rest of the things in the `Troubleshooting` section. I think only core-devs have access to manipulate workflows.

**Comment by nilsdeppe** [comment]
Added notes that the buttons and workflows may only be run if you have write access to the repo. Thanks for the review, @knelli2!

**Review by knelli2 — DISMISSED** [review]

File: docs/DevGuide/GitHubActions.md [hunk 2]
```diff
@@ -3,41 +3,36 @@ Distributed under the MIT License.
...
-succeeded` in the section for `All checks have failed` or `All checks
-have passed` towards the bottom of the pull request.  You can login to
-Travis using your GitHub credentials in order to cancel your builds,
-and you should do so if you update your pull request while it is
-building.
+each time a pull request is submitted or updated.  GitHub Actions will also
+launch these build jobs each time you push to a branch on your fork of SpECTRE
+if you enable it. GitHub Actions is also used to deploy releases of the code.
+
+For pull requests, you can view the GitHub Actions CI build by clicking on
```
Comment (kidder) [inline docs/DevGuide/GitHubActions.md]: on the

File: docs/DevGuide/GitHubActions.md [hunk 3]
```diff
@@ -84,29 +79,33 @@ if any of the following are true:
   - The file `tests/Unit/RunSingleTest/CMakeLists.txt` wasn't modified
   - A python file is not formatted according to the `.style.yapf` file in the
     root of the repository.
-* RUN_CLANG_TIDY runs the script `.travis/RunClangTidy.sh` which runs
-`clang-tidy` on all files which were modified.  This is done for both `Release`
-and `Debug` builds.
+* RUN_CLANG_TIDY runs clang-tidy on the source code  This is done for both
```
Comment (kidder) [inline docs/DevGuide/GitHubActions.md]: add period after source code

File: docs/DevGuide/GitHubActions.md [hunk 4]
```diff
@@ -165,24 +177,10 @@ specifying the environment variables:
 \code{.sh}
 CCACHE_COMPILERCHECK=content
 CCACHE_EXTRAFILES="${SPECTRE_SOURCE_DIR}/tools/SpectrePch.hpp"
-CCACHE_IGNOREHEADERS="${SPECTRE_BUILD_DIR}/SpectrePch.hpp:${SPECTRE_BUILD_DIR}/SpectrePch.hpp.gch"
+CCACHE_IGNOREHEADERS=\
```
Comment (kidder) [inline docs/DevGuide/GitHubActions.md:180]: the `$` in these environment variables are being rendered as `\f$`

File: docs/DevGuide/GitHubActions.md [hunk 5]
```diff
@@ -165,24 +177,10 @@ specifying the environment variables:
...
-
-In order to avoid timeouts we build SpECTRE in various stages, carrying over the
-ccache from one stage to the next. This allows us to avoid recompiling the code
-in the next stage (there is some small overhead from running ccache instead of
-not doing anything at all). The first stage builds all the SpECTRE libraries but
-none of the executables or testing libraries. The second stage builds builds the
-test executables, runs the tests, and also runs ClangTidy, include-what-you-use,
-and various other checks. Another stage could be added that builds some of the
-test libraries if necessary.
-
```
Comment (kidder) [inline docs/DevGuide/GitHubActions.md:185]: another mis-rendered `$` below (for mac_cache, but curiously the one for cache renders correctly)

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash immediately

File: docs/DevGuide/GitHubActions.md (hunk 4 above)
Comment (nilsdeppe) [reply docs/DevGuide/GitHubActions.md:180]: See #4036

**Review by nilsdeppe — COMMENTED** [review]
Rebased and squashed

File: docs/DevGuide/GitHubActions.md (hunk 4 above)
Comment (nilsvu) [reply docs/DevGuide/GitHubActions.md:180]: This might render correctly if you replace `\code` with three backticks. Either way, I'll take a look.

**Review by kidder — APPROVED** [review]

**Pull request merged**